scripts/biomedical/NCBI_Taxonomy/scripts/format_ncbi_taxonomy.py

```diff
@@ -0,0 +1,656 @@
+            mcf_file_path (str): Enum MCF file path
+        """
+        unique_host_lst = df.host.unique()
+        set_unique_host = set()
+        for uh in unique_host_lst:
+            unique_host = uh.split(',')
+            for u in unique_host:
+                set_unique_host.add(u)
+
+        sorted_lst = sorted(set_unique_host)
+        with open(mcf_file_path, 'a') as file:
+            file.write(HOST_ENUM_MCF)
+            for uh in sorted_lst:
+                biological_case_item = string.capwords(uh).replace(" ", "")
+                mcf_line = HOST_MCF.format(
+                    biological_case_item=biological_case_item, item=uh)
+                file.write(mcf_line)
+
+    def update_host_enum(self, df: pd.DataFrame) -> None:
+        """Update Host Enum in FINAL_NCBI_TAXONOMY based on tax_id
+
+        Args:
+            df (pd.DataFrame): Host DataFrame
+        """
+        global FINAL_NCBI_TAXONOMY
+        for _, row in df.iterrows():
+            try:
+                current_node = FINAL_NCBI_TAXONOMY[row['tax_id']]
+                current_node['host'] = row['DC_host_enum']
+            except:
+                logging.error(
+                    f"Host tax_id -  {row['tax_id']} is missing in FINAL_NCBI_TAXONOMY"
+                )
+
+
+class NodesCls:
+    """Class to process nodes.dmp file
+    """
+
+    def create_dataframe(self) -> pd.DataFrame:
+        """Creates Nodes DataFrame
+
+        Returns:
+            pd.DataFrame: Nodes DataFrame
+        """
+        df = DataFrameGenerator().get_dataframe(SOURCE_FILE_PATH, NODES_DMP)
+        df = DataFrameGenerator().assign_dataframe_header(df, NODES_COL)
+        df = df.drop(['embl_code'], axis=1)
+        df = df.drop(df.columns[5:], axis=1)
+        return df
+
+    def append_mcf_data(self, df: pd.DataFrame, mcf_file_path: str) -> None:
+        """Append BiologicalTaxonomicRank MCF entries to file
+
+        Args:
+            df (pd.DataFrame): Nodes DataFrame
+            mcf_file_path (str): Enum MCF file path
+        """
+        unique_rank = sorted(df['rank'].unique())
+
+        with open(mcf_file_path, 'a') as file:
+            file.write(NODES_ENUM_MCF)
+            for ur in unique_rank:
+                rank_case = string.capwords(ur).replace(" ", "")
+                mcf_line = NODES_MCF.format(rank_case=rank_case, rank=ur)
+                file.write(mcf_line)
+
+    def update_nodes_enum(self, df: pd.DataFrame) -> None:
+        """Update parentDcid, division & hasInheritedDivision in FINAL_NCBI_TAXONOMY
+
+        Args:
+            df (pd.DataFrame): Nodes DataFrame
+        """
+        global FINAL_NCBI_TAXONOMY
+        for index, row in df.iterrows():
+            try:
+                current_node = FINAL_NCBI_TAXONOMY[row['tax_id']]
+                current_node['parentDcid'] = TAX_ID_DCID_MAPPING[
+                    row["parent_tax_id"]]
+                current_node['division'] = DIVISION_DICT[row["division_code"]]
```

Comment:
The taxonRank is not processed from the "rank" column. Please add the following two lines to this function at line 514 to fix this problem:

rank_case = string.capwords(row["rank"]).replace(" ", "")
current_node['taxonRank'] = "dcs:BiologicalTaxonomicRank" + rank_case

Reply:
Code fix done